cardano_node_tests/utils/tx_view.py

```diff
@@ -183,6 +183,44 @@ def check_tx_view(  # noqa: C901
     if tx_raw_len_certs != loaded_len_certs:
         raise AssertionError(f"certificates: {tx_raw_len_certs} != {loaded_len_certs}")
 
+    certificates_information = {
+        "genesis key delegation": ["VRF key hash", "delegate key hash", "genesis key hash"],
+        "MIR": ["pot", "target stake addresses", "send to treasury", "send to reserves"],
+        "stake address deregistration": [
+            "stake credential key hash",
+            "stake credential script hash",
+        ],
+        "stake address registration": ["stake credential key hash", "stake credential script hash"],
+        "stake address delegation": [
+            "pool",
+            "stake credential key hash",
+            "stake credential script hash",
+        ],
+        "stake pool retirement": ["epoch", "pool"],
+        "stake pool registration": [
+            "VRF key hash",
+            "cost",
+            "margin",
+            "metadata",
+            "owners (stake key hashes)",
+            "pledge",
+            "pool",
+            "relays",
+            "reward account",
+        ],
+    }
+
+    for certificate in tx_loaded.get("certificates") or []:
+        certificate_name = list(certificate.keys())[0]
+        certificate_fields = list(list(certificate.values())[0].keys())
+
+        if certificates_information.get(certificate_name) and not all(
+            elem in certificates_information[certificate_name] for elem in certificate_fields
```

Comment:
It would be more readable and faster to have the values of `certificates_information` changed to sets (instead of lists), make `certificate_fields` a set as well, and use `certificate_fields.issubset(certificates_information[certificate_name])`